source/Octopus.Tentacle.Client/Scripts/ScriptExecutionOrchestrator.cs

```diff
@@ -27,15 +29,17 @@ internal class ScriptExecutionOrchestrator : IDisposable
         readonly Func<CancellationToken, Task> onScriptCompleted;
         readonly TimeSpan onCancellationAbandonCompleteScriptAfter;
         readonly RpcRetrySettings rpcRetrySettings;
+        readonly AsyncHalibutFeature asyncHalibutFeature;
         readonly ILog logger;
-        readonly IClientScriptService scriptServiceV1;
-        readonly IClientScriptServiceV2 scriptServiceV2;
-        readonly IClientCapabilitiesServiceV2 capabilitiesServiceV2;
+        
+        readonly SyncAndAsyncClientScriptServiceV1 clientScriptServiceV1;
+        readonly SyncAndAsyncClientScriptServiceV2 clientScriptServiceV2;
+        readonly SyncAndAsyncClientCapabilitiesServiceV2 clientCapabilitiesServiceV2;
 
         public ScriptExecutionOrchestrator(
```

Comment:
This class has become very bloated and could do with some refactoring to break it apart. For another PR